src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1083,12 +1083,24 @@ private IEnumerable<TypeInferenceInfo> InferTypeInCatchFilterClause(CatchFilterC
                     return CreateResult(SpecialType.System_Object);
                 }
 
-                // TODO: mark the reference case with a ? once we can check if we're within a nullable context
-                // (tracked by https://github.com/dotnet/roslyn/issues/36101)
                 return rightTypes
-                    .Select(x => x.InferredType.IsValueType
-                                     ? new TypeInferenceInfo(this.Compilation.GetSpecialType(SpecialType.System_Nullable_T).Construct(x.InferredType.WithoutNullability())) // Goo() ?? 0
-                                     : x); // Goo() ?? ""
+                    .Select(x => new TypeInferenceInfo(MakeNullable(x.InferredType))); // Goo() ?? ""
+
+                ITypeSymbol MakeNullable(ITypeSymbol symbol)
```

Comment:
make static, pass compilation in?

Reply:
Good idea, if only we had our refactoring for that done yet...